timemachine/fe/single_topology.py

```diff
@@ -796,102 +826,50 @@ def find_dummy_groups_and_anchors(
-    to the interval [lambda_min, lambda_max], and pin to the end state values outside of this range.
+    If lamb <= lambda_min, src_k is returned.
+    If lamb >= lambda_min, and lamb <= lambda_max, interpolate_fn(src_k, dst_k, lamb) is returned.
+    If lamb >= lambda_max, dst_k is returned.
 
     Parameters
     ----------
-    f : callable, (src_k: float, dst_k: float, lam: float) -> float
-        interpolation function; should satisfy f(0) = src_k, f(1) = dst_k
+    interpolate_fn: f(src_params, dst_params, lamb) -> param
+        interpolation function operating over open interval (0,1). Boundary values are discarded.
 
     src_k, dst_k : float, k >= 0
         force constants at lambda=0 and lambda=1, respectively
 
-    lambda_min, lambda_max : float, in 0 < lambda_min < lambda_max < 1
-        interpolate in range [lambda_min, lambda_max] (pin to end states otherwise). Note that if dst_k=0, the
-        convention is flipped so that 1 - lambda_min corresponds to f(0) and 1 - lambda_max corresponds to f(1).
-
-    Returns
-    -------
-    float
-        interpolated force constant
-    """
-
-    def ring_closing(dst_k, lamb):
-        return interpolate.pad(f, 0.0, dst_k, lamb, lambda_min, lambda_max)
+    lamb: float
+        interpolation value, lamb in [0, 1]
 
-    def ring_opening(src_k, lamb):
-        return ring_closing(src_k, 1.0 - lamb)
-
-    return jnp.where(
-        src_k == 0.0,
-        ring_closing(dst_k, lamb),
-        jnp.where(
-            dst_k == 0.0,
-            ring_opening(src_k, lamb),
-            f(src_k, dst_k, lamb),
-        ),
-    )
-
-
-def interpolate_harmonic_force_constant(src_k, dst_k, lamb, k_min, lambda_min, lambda_max):
-    """
-    Interpolate between force constants using a log-linear functional form.
-
-    In the special case when src_k=0 or dst_k=0 (e.g. ring opening or closing transformations):
-
-    1. Intermediates are interpolated from k_min instead of zero (since 0 is not in the range of the interpolation
-       function)
-    2. Interpolation is restricted to the interval [lambda_min, lambda_max] and pinned to the end state values outside
-       of this range
-
-    Parameters
-    ----------
-    src_k, dst_k : float, k >= 0
-        force constants at lambda=0 and lambda=1, respectively
-
-    k_min : float, k_min > 0
-        minimum force constant for interpolation
-
-    lambda_min, lambda_max : float, in 0 < lambda_min < lambda_max < 1
-        interpolate in range [lambda_min, lambda_max] (pin to end states otherwise). Note that if dst_k=0, the
-        convention is flipped so that 1 - lambda_min corresponds to f(0) and 1 - lambda_max corresponds to f(1).
+    lambda_min, lambda_max : float, in 0 <= lambda_min < lambda_max <= 1
+        boundaries for interpolation. See docstring above on expected behavior.
 
     Returns
     -------
     float
         interpolated force constant
     """
-
     return jnp.where(
-        lamb == 0.0,
+        lamb <= lambda_min,
```

Comment:
The `where` branches here seem redundant with the logic in `intepolate.pad`, unless I'm missing something?

Reply:
Ah, I think you're right - I removed some symmetrization code in an earlier version that made it obsolete.

Reply:
fixed by deletion